### PR TITLE
Daily survey results

### PR DESCRIPTION
**Pull request by mehalshah**
- [ ] Basic API for summarizing daily surveys. We don't have the translation layer done yet, so just faking the data for the time being
- [ ] Basic amount of UI
- [ ] Basic component testing

### CODE REVIEW

**Comment by mehalshah** [comment]
image

File: apps/src/code-studio/pd/workshop_dashboard/reports/local_summer_workshop_daily_survey/results_loader.jsx [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+    if (this.state.loading) {
+      return (
+        <div>
+          <Spinner/>
+        </div>
+      );
+    } else {
+      return (
+        <Results
+          {..._.omit(this.state, ['loading'])}
```
Comment (aoby) [inline apps/src/code-studio/pd/workshop_dashboard/reports/local_summer_workshop_daily_survey/results_loader.jsx]: note, this omit can be done in raw ES6 by adding the following to the beginning of the function:
```jsx
const {loading, ...data} = this.state;
// Now we have 2 local vars, loading and data, where `data` contains all the state except `loading`
```

File: dashboard/app/helpers/pd/workshop_survey_results_helper.rb [hunk 2]
```diff
@@ -159,4 +159,66 @@ def summarize_workshop_surveys(workshops:, facilitator_name_filter: nil, facilit
...
+        puts surveys.map {|survey| survey[q_key]}
+        sum = surveys.map {|survey| survey[q_key]}.reduce(0, :+)
+        session_summary[q_key] = (sum / surveys.count.to_f).round(2)
+      end
+    end
+
+    session_summary
+  end
+
+  def get_surveys_for_sessions(sessions)
```
Comment (aoby) [inline dashboard/app/helpers/pd/workshop_survey_results_helper.rb]: Please add comments to make it clear this is (temporary) dummy data

File: dashboard/app/helpers/pd/workshop_survey_results_helper.rb [hunk 3]
```diff
@@ -159,4 +159,66 @@ def summarize_workshop_surveys(workshops:, facilitator_name_filter: nil, facilit
...
+    questions = get_questions_for_form
+    session_summary = {
+      attending_teachers: sessions.map(&:attendances).map(&:count).reduce(:+),
+      completed_surveys: surveys.count
+    }
+    questions.each do |q_key, question|
+      if include_free_response && question[:free_response]
+        sum = surveys.map {|survey| survey[q_key]}.reduce([], :append)
+        session_summary[q_key] = sum
+      elsif question[:free_response].nil?
```
Comment (aoby) [inline dashboard/app/helpers/pd/workshop_survey_results_helper.rb]: why `nil?` specifically? What if we specify `free_response: false`?

File: dashboard/app/helpers/pd/workshop_survey_results_helper.rb [hunk 4]
```diff
@@ -159,4 +159,66 @@ def summarize_workshop_surveys(workshops:, facilitator_name_filter: nil, facilit
...
+        puts surveys.map {|survey| survey[q_key]}
```
Comment (aoby) [inline dashboard/app/helpers/pd/workshop_survey_results_helper.rb]: leftover debugging?

**Comment by tanyaparker** [comment]
[A Regional Partner can see the results of each local daily survey in a separate tab](https://trello.com/c/hHBWgcS6/600-a-regional-partner-can-see-the-results-of-each-local-daily-survey-in-a-separate-tab)

**Comment by tanyaparker** [comment]
[A Regional Partner can see the results of each local daily survey in a separate tab](https://trello.com/c/hHBWgcS6/600-a-regional-partner-can-see-the-results-of-each-local-daily-survey-in-a-separate-tab)

File: dashboard/app/helpers/pd/workshop_survey_results_helper.rb (hunk 3 above)
Comment (mehalshah) [reply dashboard/app/helpers/pd/workshop_survey_results_helper.rb]: Restructured the conditionals.

**Pull request merged**